packages/core/src/ctype/CType.ts

```diff
@@ -175,21 +182,8 @@ export function verifyClaimAgainstNestedSchemas(
   claimContents: Record<string, any>,
   messages?: string[]
 ): void {
-  const validator = new JsonSchema.Validator(cType, '7', false)
-  nestedCTypes.forEach((ctype) => {
-    validator.addSchema(ctype)
-  })
-  validator.addSchema(CTypeModel)
-  const { valid, errors } = validator.validate(claimContents)
-  if (valid === true) return
-  if (messages) {
-    errors.forEach((error) => {
-      messages.push(error.error)
-    })
-  }
-  throw new SDKErrors.NestedClaimUnverifiableError(undefined, {
-    cause: errors,
-  })
+  verifyObjectAgainstSchema(cType, CTypeModel, messages)
```

Comment:
Question, probably out of scope for this PR: how is a verifier supposed to know what are the nested CTypes? Are they assumed to have this knowledge? Or might they only know the top-level CType they want to verify a credential against?

Reply:
I was thinking the same thing but yes, I think it's out of scope. Maybe we need a CType registry in the sdk? Would also let us validate a CType only once (when adding to the registry) and not every time it is used.

Reply:
That would make our SDK less stateless, which is something we wanted to strive for as much as possible. it might be a good idea to at least reference this in a comment or something.

Reply:
Guess that's true. The way I see it choosing a CType is a very conscious decision for a verifier, so they'd be familiar with it and know whether or not it is nested. At least the need to known what sort of data they will be working with in their application, so they'd look into that for sure.